Metro-Records/la-metro-councilmatic
# Refactor SmartLogic interactions

## Overview

This PR takes most of the code from #771, omitting logic to filter topics without any associated bills. In short, it:

- Creates a new, separate `smartlogic` app containing a SmartLogic client and Django views to retrieve relations and suggestions. These views return the raw responses from the SES API.
- Adds a LA Metro SmartLogic API view, which subclasses the upstream view from the `smartlogic` app, processes the responses, and returns only subjects we have in our database.
- Refactors the autocomplete and related terms JavaScript to use our endpoints, instead querying SES directly. This allows us to move that complex processing out of JavaScript, yahoo!

Connects #975

### Notes

Behavior should not change.

![metro-search](https://github.com/Metro-Records/la-metro-councilmatic/assets/12176173/48b0067f-3fe2-4c82-a239-65af7d58d0f7)

## Testing Instructions

 * Configure your local app to use the v5 Test API. The environment and API key can be found by logging into cloud.smartlogic.com using the credentials in Bitwarden.
 * Retrieve subject GUIDs and classifications: `docker-compose run --rm app python manage.py refresh_guid`
 * Update your search index: `docker-compose run --rm app python manage.py update_index --batch-size=25`
 * Bring your app up (`docker-compose up app`), then navigate to the homepage and confirm autosuggestions and term selection work as expected in the search bar.
 * Run a search and confirm related terms are populated beneath the search filters.
